src/Settings.php

```diff
@@ -23,6 +23,8 @@ abstract class Settings implements Arrayable, Jsonable, Responsable, Serializabl
 
     private bool $configInitialized = false;
 
+    protected $originalSettingProperties;
```

Comment:
Can you rename this to `originalValues`? And type it as `?Collection` with `null` as default?